framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -1210,4 +1212,30 @@ private void addAnnotationToStore(CFStore store, AnnotationMirror anno, Node nod
                 resultValues,
                 transferResult.getResultValue().getUnderlyingType());
     }
+
+    /**
+     * The value checker doesn't override {@link
+     * org.checkerframework.framework.flow.CFAbstractTransfer#strengthenAnnotationOfEqualTo(TransferResult,
+     * Node, Node, CFAbstractValue, CFAbstractValue, boolean)}, which is called by {@link
+     * org.checkerframework.framework.flow.CFAbstractTransfer#visitCase(CaseNode, TransferInput)}.
+     * Therefore, this needs to explicitly call the handling for binary comparison to do the extra
+     * refinements that the Value Checker applies on switch statements.
+     */
+    @Override
+    public TransferResult<CFValue, CFStore> visitCase(
+            CaseNode n, TransferInput<CFValue, CFStore> p) {
+        TransferResult<CFValue, CFStore> transferResult = super.visitCase(n, p);
+        // TODO: this should work for chars too, but it is causing a crash when building the
```

Comment:
I think the correct thing to do is make the Value Checker override strengthenAnnotationOfEqualTo.  If the types are numerical, call calculateBinaryComparison; otherwise, it should refine the left and right hand sides to the intersection of the known values.  I'll work on this.